adserver/api/views.py

```diff
@@ -148,43 +151,52 @@ def _prepare_response(self, ad, placement, publisher, keywords, forced=False):
 
         Data passed to `offer_ad` is cached for use on the View & Click tracking.
         """
-        # Record a decision for every call to the API
-
         ad_type_slug = placement.get("ad_type")
         div_id = placement.get("div_id")
 
-        if not ad:
-            Advertisement.record_null_offer(
+        # Check if this client should get a sticky ad decision
+        cache_key = self._sticky_decision_cache_key(publisher, ad_type_slug)
+        data = cache.get(cache_key)
```

Comment:
I think we should probably log here if we're using a cached response, so we have a bit more data around this. INFO level probably to start, then we can drop it to DEBUG?

Reply:
Added